pkg/client/cache/reflector_test.go

```diff
@@ -36,8 +36,8 @@ type testLW struct {
 }
 
 func (t *testLW) List() (runtime.Object, error) { return t.ListFunc() }
-func (t *testLW) Watch(resourceVersion string) (watch.Interface, error) {
-	return t.WatchFunc(resourceVersion)
+func (t *testLW) Watch(options api.ListOptions) (watch.Interface, error) {
+	return t.WatchFunc(options.ResourceVersion)
```

Comment:
Would like to see a reflector test exercising the force resync, watching timeout, and resync loop behavior.